app/services/customers/update_service.rb

```diff
@@ -75,7 +76,14 @@ def update(**args)
       customer.external_id = args[:external_id] if customer.editable? && args.key?(:external_id)
 
       ActiveRecord::Base.transaction do
+        # TODO:
+        # if args[:payment_provider].nil? && args[:payment_provider_code].present?
```

Comment:
I would just add one extra condition: `if old_provider_customer && args[:payment_provider].nil? && args[:payment_provider_code].present?`, to prevent UndefinedMethod on nil class errors, just in case